pkg/apis/extensions/types.go

```diff
@@ -899,3 +899,45 @@ type NetworkPolicyList struct {
 
 	Items []NetworkPolicy `json:"items"`
 }
+
+// +genclient=true
+// +nonNamespaced=true
+
+// StorageClass describes the parameters for a class of storage for
+// which PersistentVolumes can be dynamically provisioned.
+//
+// StorageClasses are non-namespaced; the name of the storage class
+// according to etcd is in ObjectMeta.Name.
+type StorageClass struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	// ProvisionerType indicates the type of the provisioner.
+	ProvisionerType string `json:"provisionerType,omitempty"`
+
+	// ProvisionerParameters holds the parameters for the provisioner that should
+	// create volumes of this storage class.
+	ProvisionerParameters map[string]string `json:"provisionerParameters,omitempty"`
```

Comment:
@rootfs @colemickens 

If you need a secret, the coordinates of the secret should be a parameter to the provisioner.  Let me note -- this would be the first case in which we allowed secrets to be referenced from outside their namespace and will need to be thought through from an auth/ACL/permissions perspective.